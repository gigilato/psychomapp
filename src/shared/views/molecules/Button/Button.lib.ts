/* eslint-disable @typescript-eslint/strict-boolean-expressions */
import clsx from 'clsx'

import { spacing } from '$theme'

import { IButtonProps } from './Button.props'

export const ICON_SIZE = spacing.l
export const HIT_SLOP = 5
export const ERROR_OPACITY = 0.2

export const useButton = ({
  variant,
  disabled,
  error,
  className,
  outlined,
}: Pick<IButtonProps, 'variant' | 'disabled' | 'error' | 'outlined' | 'className'>) => {
  const containerClassName = clsx(
    'flex-row items-center justify-center rounded-s',
    variant === 'primary' ? 'h-[44]' : 'h-[40]',
    variant === 'primary' ? 'bg-primary-classic' : 'bg-secondary-strong',
    disabled && 'bg-disabled border-disabled border-[2px]',
    outlined && 'bg-white-light border-[1px]',
    outlined && variant === 'primary' && 'border-primary-classic',
    outlined && variant === 'secondary' && 'border-secondary-strong',
    error && 'bg-white-light border-danger border-[2px]',
    className
  )

  return containerClassName
}

const getTextColor = ({
  variant,
  disabled,
  error,
  outlined,
}: Pick<IButtonProps, 'disabled' | 'error' | 'outlined' | 'variant'>) => {
  if (disabled) return 'text-grey-strong'
  if (error) return 'text-danger'
  if (outlined) return variant === 'primary' ? 'text-primary-classic' : 'text-secondary-strong'
  return 'text-white-light'
}
export const useTextButton = (props: Pick<IButtonProps, 'disabled' | 'error' | 'outlined'>) => {
  return getTextColor(props)
}
