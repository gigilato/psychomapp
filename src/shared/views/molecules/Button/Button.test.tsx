import { render } from '@testing-library/react-native'

import { IconButton, Button, PressableIcon, PressableText } from './Button'

describe('Button', () => {
  it('renders correctly', async () => {
    const component = render(<Button ID="" title="Title" />)
    expect(component).toMatchSnapshot()
  })
  it('renders loading correctly', async () => {
    const component = render(<Button ID="" title="Title" isLoading />)
    expect(component).toMatchSnapshot()
  })
  it('renders disabled correctly', async () => {
    const component = render(<Button ID="" title="Title" disabled />)
    expect(component).toMatchSnapshot()
  })
  it('renders error correctly', async () => {
    const component = render(<Button ID="" title="Title" error />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon right correctly', async () => {
    const component = render(<Button ID="" title="Title" iconRight="bell" />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon left correctly', async () => {
    const component = render(<Button ID="" title="Title" iconLeft="bell" />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon right disabled correctly', async () => {
    const component = render(<Button ID="" title="Title" iconRight="bell" disabled />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon left disabled correctly', async () => {
    const component = render(<Button ID="" title="Title" iconLeft="bell" disabled />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon right error correctly', async () => {
    const component = render(<Button ID="" title="Title" iconRight="bell" error />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon left error correctly', async () => {
    const component = render(<Button ID="" title="Title" iconLeft="bell" error />)
    expect(component).toMatchSnapshot()
  })
})

describe('IconButton', () => {
  it('renders correctly', async () => {
    const component = render(<IconButton ID="" icon="bell" />)
    expect(component).toMatchSnapshot()
  })
  it('renders loading correctly', async () => {
    const component = render(<IconButton ID="" icon="bell" isLoading />)
    expect(component).toMatchSnapshot()
  })
  it('renders disabled correctly', async () => {
    const component = render(<IconButton ID="" icon="bell" disabled />)
    expect(component).toMatchSnapshot()
  })
  it('renders error correctly', async () => {
    const component = render(<IconButton ID="" icon="bell" error />)
    expect(component).toMatchSnapshot()
  })
})

describe('PressableText', () => {
  it('renders correctly', async () => {
    const text = 'text'
    const component = render(
      <PressableText ID="test" textClassName="text-error" onPress={jest.fn()}>
        {text}
      </PressableText>
    )
    component.getByText(text)
    expect(component).toMatchSnapshot()
  })
})

describe('PressableIcon', () => {
  it('renders correctly', async () => {
    const component = render(
      <PressableIcon
        ID="test"
        className="bg-error"
        icon="anchor"
        iconClassName="text-primary-classic"
        onPress={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
