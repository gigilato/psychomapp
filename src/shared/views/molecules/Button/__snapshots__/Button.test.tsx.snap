// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Title
  </Text>
</View>
`;

exports[`Button renders disabled correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="none"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "backgroundColor": "#E1DCDC",
      },
      {
        "borderBottomColor": "#E1DCDC",
        "borderLeftColor": "#E1DCDC",
        "borderRightColor": "#E1DCDC",
        "borderTopColor": "#E1DCDC",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      [
        {},
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#505050",
        },
      ]
    }
  >
    Title
  </Text>
</View>
`;

exports[`Button renders error correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  animationOpacity={0.2}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "borderBottomColor": "#AE3232",
        "borderLeftColor": "#AE3232",
        "borderRightColor": "#AE3232",
        "borderTopColor": "#AE3232",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#AE3232",
        },
      ]
    }
  >
    Title
  </Text>
</View>
`;

exports[`Button renders icon left correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#FFFFFF",
          },
          {
            "marginRight": 8,
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Title
  </Text>
</View>
`;

exports[`Button renders icon left disabled correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="none"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "backgroundColor": "#E1DCDC",
      },
      {
        "borderBottomColor": "#E1DCDC",
        "borderLeftColor": "#E1DCDC",
        "borderRightColor": "#E1DCDC",
        "borderTopColor": "#E1DCDC",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      [
        {},
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#505050",
          },
          {
            "marginRight": 8,
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#505050",
        },
      ]
    }
  >
    Title
  </Text>
</View>
`;

exports[`Button renders icon left error correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  animationOpacity={0.2}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "borderBottomColor": "#AE3232",
        "borderLeftColor": "#AE3232",
        "borderRightColor": "#AE3232",
        "borderTopColor": "#AE3232",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#AE3232",
          },
          {
            "marginRight": 8,
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#AE3232",
        },
      ]
    }
  >
    Title
  </Text>
</View>
`;

exports[`Button renders icon right correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Title
  </Text>
  <Text />
</View>
`;

exports[`Button renders icon right disabled correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="none"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "backgroundColor": "#E1DCDC",
      },
      {
        "borderBottomColor": "#E1DCDC",
        "borderLeftColor": "#E1DCDC",
        "borderRightColor": "#E1DCDC",
        "borderTopColor": "#E1DCDC",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      [
        {},
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#505050",
        },
      ]
    }
  >
    Title
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#505050",
          },
          {
            "marginLeft": 8,
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Button renders icon right error correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  animationOpacity={0.2}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "borderBottomColor": "#AE3232",
        "borderLeftColor": "#AE3232",
        "borderRightColor": "#AE3232",
        "borderTopColor": "#AE3232",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Bold",
        },
        {
          "fontSize": 16,
        },
        {
          "color": "#AE3232",
        },
      ]
    }
  >
    Title
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#AE3232",
          },
          {
            "marginLeft": 8,
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Button renders loading correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <ActivityIndicator
    color="white"
  />
</View>
`;

exports[`IconButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "width": 56,
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#FFFFFF",
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`IconButton renders disabled correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="none"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "backgroundColor": "#E1DCDC",
      },
      {
        "borderBottomColor": "#E1DCDC",
        "borderLeftColor": "#E1DCDC",
        "borderRightColor": "#E1DCDC",
        "borderTopColor": "#E1DCDC",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      {
        "width": 56,
      },
      [
        {},
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#505050",
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`IconButton renders error correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  animationOpacity={0.2}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "borderBottomColor": "#AE3232",
        "borderLeftColor": "#AE3232",
        "borderRightColor": "#AE3232",
        "borderTopColor": "#AE3232",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      {
        "width": 56,
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#AE3232",
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`IconButton renders loading correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="scale"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "height": 44,
      },
      {
        "backgroundColor": "#FFB300",
      },
      {
        "width": 56,
      },
      [
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <ActivityIndicator
    color="white"
  />
</View>
`;

exports[`PressableIcon renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={5}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "opacity": undefined,
        },
        undefined,
      ],
    ]
  }
  testID="test"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "color": "#FFB300",
          },
        ],
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`PressableText renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={5}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "opacity": undefined,
        },
        undefined,
      ],
    ]
  }
  testID="test"
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
      ]
    }
  >
    text
  </Text>
</View>
`;
